web/src/Root.tsx

```diff
@@ -0,0 +1,672 @@
+import ClusterNodes from "./components/apps/ClusterNodes";
+import UnsupportedBrowser from "./components/static/UnsupportedBrowser";
+import NotFound from "./components/static/NotFound";
+import { Utilities, parseUpstreamUri } from "./utilities/utilities";
+import fetch from "./utilities/fetchWithTimeout";
+import SecureAdminConsole from "./components/SecureAdminConsole";
+import UploadLicenseFile from "./components/UploadLicenseFile";
+import BackupRestore from "./components/BackupRestore";
+import UploadAirgapBundle from "./components/UploadAirgapBundle";
+import RestoreCompleted from "./components/RestoreCompleted";
+import Access from "./components/identity/Access";
+import SnapshotsWrapper from "./components/snapshots/SnapshotsWrapper";
+import { QueryClient, QueryClientProvider } from "react-query";
+import { InstallWithHelm } from "@features/AddNewApp/components/InstallWithHelm";
+import { RouteProps } from "react-router";
+
+import Footer from "./components/shared/Footer";
+import NavBar from "./components/shared/NavBar";
+
+import "./scss/index.scss";
+import connectHistory from "./services/matomo";
+
+// react-query client
+const queryClient = new QueryClient();
+
+const INIT_SESSION_ID_STORAGE_KEY = "initSessionId";
+
+let browserHistory = createBrowserHistory();
+let history = connectHistory(browserHistory);
+
+// TODO:  pull in the react router hook
+const ProtectedRoute = ({
+  render,
+  location,
+  path,
+}: {
+  render: (props: object) => React.ReactNode;
+} & RouteProps) => {
+  const redirectURL = `/secure-console?next=${location?.pathname}${location?.search}`;
+
+  return (
+    <Route
+      path={path}
+      render={(innerProps) => {
+        if (Utilities.isLoggedIn()) {
+          return render(innerProps);
+        }
+        return <Redirect to={redirectURL} />;
+      }}
+    />
+  );
+};
+
+type ThemeState = {
+  navbarLogo: string | null;
+};
+
+const ThemeContext = React.createContext({
+  setThemeState: () => {},
+  getThemeState: (): ThemeState => ({ navbarLogo: null }),
+  clearThemeState: () => {},
+});
+
+type AppBranding = {
+  css: string;
+  fontFaces?: string[];
+  logo: string;
+};
+
+type App = {
+  slug: string;
+  isGitOpsSupported: boolean;
+  isIdentityServiceSupported: boolean;
+  isGeoaxisSupported: boolean;
+  allowSnapshots: boolean;
+};
+type State = {
+  appBranding: AppBranding | null;
+  appLogo: string | null;
+  appNameSpace: string | null;
+  appsList: App[];
+  appSlugFromMetadata: string | null;
+  adminConsoleMetadata: {
+    isKurl?: boolean;
+  };
+  connectionTerminated: boolean;
+  errLoggingOut: string;
+  featureFlags: object;
+  fetchingMetadata: boolean;
+  initSessionId: string | null;
+  isHelmManaged: boolean;
+  rootDidInitialWatchFetch: boolean;
+  selectedAppName: string | null;
+  snapshotInProgressApps: string[];
+  themeState: ThemeState;
+};
+
+let interval: ReturnType<typeof setInterval> | undefined;
+
+const Root = () => {
+  const [state, setState] = useReducer(
```

Comment:
following @alicenstar 's suggestion here. this is a lot cleaner to use for converting to stateless components